prometheus/reporter.go

```diff
@@ -0,0 +1,254 @@
+package prometheus
+
+import (
+	"crypto/sha1"
+	"fmt"
+	"net/http"
+	"sort"
+	"sync"
+	"time"
+
+	prom "github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/uber-go/tally"
+)
+
+var (
+	errorAlreadyRegistered = fmt.Errorf("Metric already registered")
+)
+
+type metricID string
+
+// Reporter is a prometheus backed tally reporter
+type Reporter struct {
+	counters  map[metricID]*prom.CounterVec
+	gauges    map[metricID]*prom.GaugeVec
+	summaries map[metricID]*prom.SummaryVec
+	mtx       sync.RWMutex
+}
+
+// HTTPHandler returns the prometheus HTTP handler for serving metrics
+func (r *Reporter) HTTPHandler() http.Handler {
+	return promhttp.Handler()
+}
+
+// NewReporter returns a new Reporter for Prometheus client backed metrics
+func NewReporter() Reporter {
```

Comment:
We should probably change the `statsd` reporter constructor in the same diff to `NewReporter` from its current name `NewStatsdReporter` to avoid the name and package stutter and be consistent with this constructor.